core-bundle/src/EventListener/DataContainer/TrackTitleSourceListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\CoreBundle\Filesystem\FilesystemUtil;
+use Contao\CoreBundle\Filesystem\VirtualFilesystem;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\DataContainer;
+use Contao\Message;
+
+#[AsCallback(table: 'tl_content', target: 'fields.textTrackSRC.load')]
+#[AsCallback(table: 'tl_content', target: 'fields.textTrackSRC.save')]
+readonly class TrackTitleSourceListener
+{
+    public function __construct(
+        private ContaoFramework $framework,
+        private VirtualFilesystem $filesStorage,
+    ) {
+    }
+
+    public function __invoke(mixed $value, DataContainer $dc): mixed
+    {
+        $invalid = [];
+
+        if (null !== $value) {
+            $fileSystemItems = FilesystemUtil::listContentsFromSerialized($this->filesStorage, $value);
+
+            foreach ($fileSystemItems as $fileSystemItem) {
+                $extraMetadata = $fileSystemItem->getExtraMetadata();
+
+                if (
+                    null === $extraMetadata->getTextTrack()?->getSourceLanguage()
+                    || '' === $extraMetadata->getLocalized()?->getFirst()?->getTitle()
+                ) {
+                    $invalid[] = $fileSystemItem->getName();
+                }
+            }
+        }
+
+        $message = $this->framework->getAdapter(Message::class);
+
+        if ([] !== $invalid) {
+            $message->addError(\sprintf($GLOBALS['TL_LANG']['ERR']['textTrackMetadataMissing'], implode(', ', $invalid)));
+        } elseif ($message->hasError()) {
+            // TODO: We need to reset all messages here as there is currently no way to show
+            // a message just for the current request.
+            $message->reset();
```

Comment:
I don‘t understand what problem exactly this solves?

Reply:
See the discussion [here](https://github.com/contao/contao/pull/7604#discussion_r1818746787) - currently our `Message` class cannot handle messages that should only be displayed for the _current_ request. `Message::add()` will always add it to the flash bag of the session, meaning it will be always also displayed for the _next_ request.

This is a workaround until we have another solution for this.